mmdet/utils/deployment/symbolic.py

```diff
@@ -292,12 +261,116 @@ def symbolic(g, input, offset, weight, stride, padding, dilation,
+    if version.parse(torch.__version__) < version.parse('1.7.1'):
+        from mmcv.ops.nms import NMSop
+        original_forward = NMSop.forward
+
+        def forward(ctx, bboxes, scores, iou_threshold, offset, score_threshold, max_num):
+            with torch.jit._disable_tracing():
+                inds = original_forward(ctx, bboxes, scores, iou_threshold, offset, score_threshold, max_num)
+            return inds
+        NMSop.forward = staticmethod(forward)
+
+
+def nms_symbolic_with_score_thr(g, bboxes, scores, iou_threshold, offset, score_threshold, max_num):
+    """
+    This function adds 'score_threshold' and 'max_output_boxes_per_class' to ONNX::NonMaxSuppression.
+    It should be removed after adding support for 'score_threshold' and 'max_num' in MMCV::NMSop.
+    """
+    from mmcv.onnx import is_custom_op_loaded
+    has_custom_op = is_custom_op_loaded()
+    if has_custom_op:
+        return g.op(
+            'mmcv::NonMaxSuppression',
+            bboxes,
+            scores,
+            iou_threshold_f=float(iou_threshold),
+            offset_i=int(offset))
+    else:
+        from torch.onnx.symbolic_opset9 import select, squeeze, unsqueeze
+        boxes = unsqueeze(g, bboxes, 0)
+        scores = unsqueeze(g, unsqueeze(g, scores, 0), 0)
+        max_output_per_class = g.op(
+            'Constant',
+            value_t=torch.tensor([max_num], dtype=torch.long))
+        iou_threshold = g.op(
+            'Constant',
+            value_t=torch.tensor([iou_threshold], dtype=torch.float))
+        score_threshold = g.op(
+            'Constant',
+            value_t=torch.tensor([score_threshold], dtype=torch.float))
+        nms_out = g.op('NonMaxSuppression', boxes, scores,
+                       max_output_per_class, iou_threshold, score_threshold)
+        return squeeze(
+            g,
+            select(
+                g, nms_out, 1,
+                g.op(
+                    'Constant',
+                    value_t=torch.tensor([2], dtype=torch.long))), 1)
+
+
+def set_extra_args_for_NMSop(score_thr, max_num):
+    """
+    This function sets new default values for 'score_threshold' and 'max_num' arguments
+    to correctly call the 'nms' operation with 4 arguments inside 'batched_nms'.
+    It should be removed after adding support for 'score_threshold' and 'max_num' in MMCV.
+    """
+    from mmcv.ops.nms import NMSop
+
+    original_forward = NMSop.forward
+    def forward(ctx, bboxes, scores, iou_threshold, offset, score_threshold=score_thr, max_num=max_num):
+        return original_forward(ctx, bboxes, scores, iou_threshold, offset, score_threshold, max_num)
+    NMSop.forward = staticmethod(forward)
+
+    original_symbolic = NMSop.symbolic
+    def symbolic(g, bboxes, scores, iou_threshold, offset, score_threshold=score_thr, max_num=max_num):
+        return original_symbolic(g, bboxes, scores, iou_threshold, offset, score_threshold, max_num)
+    NMSop.symbolic = staticmethod(symbolic)
```

Comment:
Is there a need to explicitly set the same default values for symbolic function, or actual values from the forward call will be reused automagically?

Reply:
This was necessary, I could not find a way to pass additional values of the arguments from the forward call automatically.